com.microsoft.mrtk.input/Interactors/Ray/MRTKRayInteractor.cs

```diff
@@ -172,15 +182,22 @@ public override void ProcessInteractor(XRInteractionUpdateOrder.UpdatePhase upda
                         isRelaxedBeforeSelect = false;
                     }
 
-                    if (DevicePoseSource != null && DevicePoseSource.TryGetPose(out Pose devicePose))
+                    // Use Pose Sources to calculate the interactor's pose and the attach transform's position
+                    if (AimPoseSource != null && AimPoseSource.TryGetPose(out Pose aimPose))
                     {
-                        attachTransform.rotation = PlayspaceUtilities.ReferenceTransform.rotation * devicePose.rotation;
+                        transform.SetPositionAndRotation(aimPose.position, aimPose.rotation);
+
+                        if (hasSelection)
+                        {
+                            float distanceRatio = PoseUtilities.GetDistanceToBody(aimPose) / refDistance;
+                            attachTransform.localPosition = new Vector3(initialLocalAttach.position.x, initialLocalAttach.position.y, initialLocalAttach.position.z * distanceRatio);
```

Comment:
See above comment about execution order; if the XRController updates again after this updates, then the ray attachtransform will jitter erratically as the device pose fights the aim pose

Reply:
I don't think there should be any jitter. Though it might be nice to have a dedicated execution order (i.e. controllers first and then interactors) I think it's not really noticeable unless the delta between the XRcontrollers position between any 2 frames is extremely large. The input action poses should be stable during any one frame, so in the worst case scenario, we might see a pose update one frame later due to the difference between the device poses during 2 consecutive frames being exceptionally large.

The device position only exerts some indirect control over the interactor as it moves around, but it's always relative to the previous frame, so there shouldn't be any consistent jitter when the application is running (at least, I don't think I've ever encountered any apparent jittery behavior during testing)